crates/edr_eth/src/signature/fakeable.rs

```diff
@@ -153,14 +132,7 @@ impl<SignatureT: Recoverable + Signature + PartialEq> PartialEq for Fakeable<Sig
                 FakeableData::Fake {
                     recovery_id: recovery_id2,
                 },
-            ) => {
-                // SAFETY: The address is always initialized for fake signatures.
-                let address1 = unsafe { self.address.get_unchecked() };
-                // SAFETY: The address is always initialized for fake signatures.
-                let address2 = unsafe { other.address.get_unchecked() };
-
-                recovery_id1 == recovery_id2 && address1 == address2
-            }
+            ) => recovery_id1 == recovery_id2 && self.address == other.address,
```

Comment:
Addressed in https://github.com/NomicFoundation/edr/pull/505/commits/dae9dad64c1f67091c47410fcca73499c22da383

FYI, this is slightly less performant than it could be, as the recoverable signature should guarantee that the address is also equal, but I think that's fine.